### DISCUSSION TITLE
How to apply migrations with MSSQL server?

### DISCUSSION THREAD

**Discussion by chrisknu** [hasura/graphql-engine#8180] [Question]
Hi there, 

I'm accustomed to migrations being generated with table/schema changes via a local hasura console (postgres). 

When attempting to set up migrations with a local instance of hasura/MSSQL, I'm running into this error:

hasura migrate create "init" --from-server --database-name <dbname>
FATA[0000] cannot fetch schema dump: schema dump for source <dbname>of kind mssql is not supported

I can get the metadata to export and apply just fine, but migrations seem to be failing. Any help is welcome.

**Comment by BenoitRanque** [comment]
MSSQL support for this functionality does not exist yet, you would need to create your init migration manually using the MSSQL equivalent of PG_DUMP

**Comment by chrisknu** [comment]
Is it possible to have a blank 'init' created and a 'mssql does not support
this action' warning thrown in the console? It looks like migrations after
the initial one work just fine, but it took a bit of time to figure that
out.

**Reply by BenoitRanque** [reply]
The init migration is only necessary so you can track your database's initial state before you started creating migrations with the console.
You can absolutely write that file manually. As for the errors in the console, I'm a bit confused, aren't you running into this error from the CLI?

**Comment by chrisknu** [comment]
Correct, I was hitting the “mssql is not supported” error from the cli… and
tracked it back to the line linked above. At the time, I didn’t know mssql
had no pg_dump equivalent.. and thus thought (because of the error and
missing mapping) that it was yet to be implemented. A slightly more verbose
error message would have helped short circuit me/anybody else that could
hit the same issue
